x-pack/plugins/security_solution/public/common/components/url_state/helpers.ts

```diff
@@ -185,7 +253,8 @@ export const makeMapStateToProps = () => {
           },
         },
         [CONSTANTS.timeline]: timeline,
-      },
+        [CONSTANTS.detailPanel]: detailPanel,
+      } as UrlState,
```

Comment:
Consider removing the `as UrlState` cast, as it appears to obfuscate an issue where the store can have an invalid state / shape:

The `detailPanel` is currently initialized as an empty object [on line 196](https://github.com/elastic/kibana/pull/122393/files#diff-2096f04d3a0bf2729326c452c7d9ab1cc7997c2093b8219045db817833db7b2fR196):

```tsx
let detailPanel = {};
```

however an empty object represents an invalid state, per the type definition of `CONSTANTS.detailPanel` in the `UrlState` interface:

```tsx
export interface UrlState {
  [CONSTANTS.appQuery]?: Query;
  [CONSTANTS.detailPanel]?: ToggleDetailPanel;
   // ...
}
```

Per the definition above, the detail panel must be a `ToggleDetailPanel`, or `undefined`. 

Please consider:

- Removing the `as UrlState` cast
- Adding a type declaration to  `detailPanel` and letting it initialize as `undefined`: 

```tsx
let detailPanel: ToggleDetailPanel | undefined;
```

Reply:
Updated this before I saw your comment, now I see the other comment below. Will update, thanks :)